cartofacade/carto_facade.go

```diff
@@ -178,12 +183,63 @@ type CartoFacade struct {
 	requestChan     chan Request
 }
 
-// RequestInterface defines the functionality of a WorkItem.
+// RequestInterface defines the functionality of a Request.
 // It should not be used outside of this package but needs to be public for testing purposes.
 type RequestInterface interface {
 	doWork(q *CartoFacade) (interface{}, error)
 }
 
+// Interface defines the functionality of a CartoFacade instance.
+// It should not be used outside of this package but needs to be public for testing purposes.
+type Interface interface {
```

Comment:
This is added so that we can mock `AddSensorReading` in this PR & can mock any other cartofacade public method so we can test the behavior of code that depends on cartofacade without needing to call into cartographer.

Reply:
Got it. Is there any name we can give this type that's a little more descriptive than Interface?

Reply:
I wanted to name it CartoFacadeI however that doesn't pass the go vet lint rules as the name of the package is cartofacade. External users would refer to it using cartofacade.Interface. I don't think there is a better name I can give it other than disabling that linting rule.

Reply:
got it, it seems reasonable given external callers will refer to it as cartofacade.Interface